### PR TITLE
fix: fix rollout metrics fail in unit test

### PR DESCRIPTION
**Pull request by yuki-97**
fix rollout metrics fails in `CI:L1` introduced in https://github.com/NVIDIA-NeMo/RL/pull/2917.

### CODE REVIEW

File: nemo_rl/experience/metric_utils.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import math
+import statistics
+from collections.abc import Sequence
+
+from wandb import Histogram
+
+
+def calculate_single_metric(
+    values: Sequence[float | int], batch_size: int, key_name: str
+) -> dict:
```
Comment (terrykong) [inline nemo_rl/experience/metric_utils.py:26]: [`nemo_rl/experience/metric_utils.py:24`](https://github.com/NVIDIA-NeMo/RL/blob/ca42c2dc83ed531956251211faac0b758a21897c/nemo_rl/experience/metric_utils.py#L24)

Nice clean extraction — the move is byte-for-byte faithful and the test key-set fix checks out. One small ask: `calculate_single_metric` is now a public cross-file interface (imported by [`rollout_manager.py`](https://github.com/NVIDIA-NeMo/RL/blob/ca42c2dc83ed531956251211faac0b758a21897c/nemo_rl/experience/rollout_manager.py#L28) and the unit test), but it has no docstring, while its sibling `pct` got one in this PR. It would also help to document the non-obvious denominator quirk: `/mean` divides by `batch_size`, not `len(values)`, while `/stddev` uses `len(values)`.

**Action:** add a Google-style docstring, e.g.:

```suggestion
def calculate_single_metric(
    values: Sequence[float | int], batch_size: int, key_name: str
) -> dict:
    """Compute summary statistics for a metric as slash-prefixed keys.

    Args:
        values: Per-sample metric values to aggregate.
        batch_size: Denominator for the mean. Note the mean is
            ``sum(values) / batch_size`` (NOT ``len(values)``), while
            ``stddev`` uses ``len(values)``.
        key_name: Prefix for the returned metric keys (e.g. ``"total_reward"``).

    Returns:
        Dict mapping ``"{key_name}/{stat}"`` to its value for ``stat`` in
        ``mean``, ``max``, ``min``, ``median``, ``stddev`` (``nan`` for a
        single value) and ``histogram`` (a ``wandb.Histogram``).
    """
```

File: nemo_rl/experience/metric_utils.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        f"{key_name}/mean": sum(values) / batch_size,
+        f"{key_name}/max": max(values),
+        f"{key_name}/min": min(values),
+        f"{key_name}/median": statistics.median(values),
+        f"{key_name}/stddev": statistics.stdev(values) if len(values) > 1 else math.nan,
+        f"{key_name}/histogram": Histogram(values),
+    }
+
+
+def pct(values: Sequence[float | int], p: float) -> float:
```
Comment (terrykong) [inline nemo_rl/experience/metric_utils.py:47]: [`nemo_rl/experience/metric_utils.py:37`](https://github.com/NVIDIA-NeMo/RL/blob/ca42c2dc83ed531956251211faac0b758a21897c/nemo_rl/experience/metric_utils.py#L37)

`pct` is now a public helper but has no direct unit test — it is only exercised through GPU-gated integration tests, whereas `calculate_single_metric` is covered by [`TestCalculateSingleMetric`](https://github.com/NVIDIA-NeMo/RL/blob/ca42c2dc83ed531956251211faac0b758a21897c/tests/unit/experience/test_rollouts.py#L71). Its edge cases (empty list → `0.0`, index clamping) are pure Python and cheap to cover.

**Action:** consider adding a `TestPct` class next to `TestCalculateSingleMetric` in `tests/unit/experience/test_rollouts.py` (verified passing locally, 7 passed):

```python
from nemo_rl.experience.metric_utils import pct


class TestPct:
    """Unit tests for the pct percentile helper."""

    def test_empty_returns_zero(self):
        assert pct([], 95) == 0.0

    def test_returns_float_for_int_input(self):
        result = pct([3, 1, 2], 50)
        assert isinstance(result, float)
        assert result == 2.0

    def test_sorts_unsorted_input(self):
        assert pct([5, 1, 3], 95) == 5.0

    def test_p95_small_list_clamps_to_max(self):
        assert pct([1, 2, 3, 4, 5], 95) == 5.0

    def test_p99_clamps_to_last_index(self):
        assert pct([1, 2, 3, 4, 5], 99) == 5.0

    def test_single_value(self):
        assert pct([42], 95) == 42.0

    def test_median_like_p50(self):
        assert pct([10, 20, 30, 40], 50) == 30.0
```

Not a merge blocker.

**Review by terrykong — COMMENTED** [review]
Reviewed with a team of agents (rl-expert, test-agent, bug-finder, devil's advocate). Summary:

- The fix is verified: on `main`, `run_async_nemo_gym_rollout` already emits `max_gen_tokens_per_turn/*` and `turns_per_sample/p95`,`/p99` (from #2917) but the test's expected key-set was never updated, so `test_run_async_nemo_gym_rollout` fails; this PR adds exactly the missing keys and restores manager/original parity for `test_async_rollout_manager_matches_original`.
- The helper extraction into `nemo_rl/experience/metric_utils.py` is byte-for-byte faithful, with no stale references to the deleted `_calculate_single_metric`/`_pct` anywhere in the repo, no circular imports, and correctly registered in `pyrefly.toml`. `TestCalculateSingleMetric` passes locally; all pre-commit hooks pass.
- Informational, no action needed in this PR: `max_gen_tokens_per_turn` intentionally has different key shapes across the two manager paths because each mirrors its `rollouts.py` original (3 keys for `run_async_multi_turn_rollout`, 7 for `run_async_nemo_gym_rollout`) — unifying them would break the parity test. Also, `nemo_rl/algorithms/async_utils/replay_buffer.py` still carries its own identical `_pct`; could be deduped onto `metric_utils.pct` in a follow-up.

Two optional low-severity suggestions inline (docstring, CPU-only test for `pct`).

_Generated by Claude Code_

**Comment by terrykong** [comment]
aside from ai review do you think we can add the test that was skipped in the "FAST" suite? since we know it crashed in L1, might be good to promote it to fast

File: nemo_rl/experience/metric_utils.py (hunk 1 above)
Comment (yuki-97) [reply nemo_rl/experience/metric_utils.py:26]: 34376b1

File: nemo_rl/experience/metric_utils.py (hunk 2 above)
Comment (yuki-97) [reply nemo_rl/experience/metric_utils.py:47]: 34376b1

**Comment by yuki-97** [comment]
> aside from ai review do you think we can add the test that was skipped in the "FAST" suite? since we know it crashed in L1, might be good to promote it to fast

updated in 570d3b8

**Review by terrykong — APPROVED** [review]
